contrib/compile-tests.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+#
+# Checks intended compile failures.
+
+set -euo pipefail
+
+REPO_DIR=$(git rev-parse --show-toplevel)
+
+should_pass() {
+    result=0
+    # All crates in the subdirectory pass/ must compile
+    for dir in pass/*; do
+        if [ -d "$dir" ]; then
+            pushd "$dir" > /dev/null
+            output=$(cargo build 2>&1)
```

Comment:
This shouldn't be needed but I guess it's OK. I don't expect the output to be too large.